grafana-plugin/src/models/schedule/schedule.ts

```diff
@@ -506,6 +514,34 @@ export class ScheduleStore extends BaseStore {
     });
   }
 
+  @action.bound
+  async refreshEvents() {
+    this.refreshEventsError = {};
+    const startMoment = this.rootStore.timezoneStore.calendarStartDate;
+
+    try {
+      const schedule = await this.loadItem(this.scheduleId);
+      this.rootStore.setPageTitle(schedule?.name);
+    } catch (error) {
+      runInAction(() => {
+        this.refreshEventsError = getWrongTeamResponseInfo(error);
+      });
+    }
+
+    this.updateRelatedUsers(this.scheduleId); // to refresh related users
+    await Promise.all([
+      this.updateEvents(this.scheduleId, startMoment, 'rotation'),
+      this.updateEvents(this.scheduleId, startMoment, 'override'),
+      this.updateEvents(this.scheduleId, startMoment, 'final'),
+      this.updateShiftSwaps(this.scheduleId, startMoment),
+    ]);
+  }
+
+  @action.bound
+  setScheduleId(id: string) {
+    this.scheduleId = id;
```

Comment:
hmm, why do we need to save some scheduleId in the store?

Reply:
`scheduleId` observable was already there in store and it's good, I just added action to be able to update it when a single schedule is opened.

We're still passing scheduleId as a prop and that can be changed in the future so that schedule id is retrieved from the store instead. I just didn't do it in scope of this PR

Reply:
I removed it in the end

Reply:
oh, I see, cool! I don't like an idea to store current page params like id in the store, we can get such params by react router